source/common/upstream/load_stats_reporter.cc

```diff
@@ -109,17 +118,44 @@ void LoadStatsReporter::sendLoadStatsRequest() {
         Protobuf::util::TimeUtil::MicrosecondsToDuration(
             std::chrono::duration_cast<std::chrono::microseconds>(measured_interval).count()));
     clusters_[cluster_name] = now;
+    auto metric_name = cluster.info()->loadReportRouterStats().http_upstream_rq_time_.name();
+    relevant_metrics.insert(metric_name);
+    metrics_to_cluster_map.insert({metric_name, cluster_stats});
   }
 
-  Config::VersionConverter::prepareMessageForGrpcWire(request_, transport_api_version_);
-  ENVOY_LOG(trace, "Sending LoadStatsRequest: {}", request_.DebugString());
-  stream_->sendMessage(request_, false);
-  stats_.responses_.inc();
-  // When the connection is established, the message has not yet been read so we
-  // will not have a load reporting period.
-  if (message_) {
-    startLoadReportPeriod();
-  }
+  internal_stats_handler_->receiveGlobalStats(
+      relevant_metrics,
+      [&](std::multimap<std::string, const Envoy::Stats::HistogramStatistics&> histogram_stats)
+          -> void {
+        for (auto entry : histogram_stats) {
+          if (metrics_to_cluster_map.find(entry.first) != metrics_to_cluster_map.end()) {
+            auto* cluster_stats = metrics_to_cluster_map[entry.first];
+            auto supported_quantiles = entry.second.supportedQuantiles();
+            auto computed_quantiles = entry.second.computedQuantiles();
+            for (size_t i = 0; i < supported_quantiles.size(); i++) {
+              double val = computed_quantiles[i];
+              if (std::isnan(val)) {
+                val = 0;
+              }
+              auto num = ProtobufWkt::DoubleValue();
+              num.set_value(val);
+              cluster_stats->mutable_request_latency_percentiles()->insert(
+                  Protobuf::MapPair<std::string, ProtobufWkt::DoubleValue>(
+                      std::to_string(supported_quantiles[i]), num));
```

Comment:
Does this cover only the last load reporting interval?

Reply:
No. It covers the last "metrics reporting interval". My understanding is that every metrics period, `ParentHistogramImpl` will merge all the thread local histograms and flush them to sinks. That's the largest source of potential confusion for a user here.

The only way I can think of implementing it for last load reporting interval is by implementing a completely different `StatsRoot` in `server.cc` which the `InternalStatsHandler` merges after we request data. I think that will be the most correct way of doing this. If this is fine with you, I can implement that.

Reply:
That makes more sense to me. @jmarantz @mattklein123 WDYT? I don't have the full histogram history context.

Reply:
> The only way I can think of implementing it for last load reporting interval is by implementing a completely different StatsRoot in server.cc which the InternalStatsHandler merges after we request data. I think that will be the most correct way of doing this. If this is fine with you, I can implement that.

Yes, if we want an independent merging period, it definitely needs to be an independent merge operation. However, I would like to make sure we make this somewhat generic as independent merge intervals has come up before. Stepping back can you review this PR with @jmarantz? I haven't looked at this in detail.

Reply:
I am not sure I can wrap my arms around the implications of having a second ThreadLocalStore instance.

Could we instead define groups of histograms that get merged at independent time-intervals? It wouldn't seem too crazy have each histogram be, by default, in a DefaultGroup, and have a histogram API that moves it into a custom HistogramGroup.

Reply:
@jmarantz okay. would you be okay with us passing the `StoreRoot` directly to the `LoadStatsReporter`, or would you like an accessor interface for accessing values for the custom `HistogramGroup`

Reply:
I feel like an accessor interface might be better but before you invest more I'd like to hear from @mattklein123 @snowp @ggreenway  if they like this direction. It will be a fair amount of work to add the new APIs as there's lots of implementations of every interface.

Reply:
+1 I would prefer to not pass the store root around if possible.

Reply:
@jmarantz would it make sense to only define a `HistogramGroup`, or a more general `MetricGroup`? There might be cases where we want to merge some counters in a separate merge interval too. 

What do you think of creating an Enum of `HistogramGroup`/`MetricGroup`? It doesn't seem like there need to be too many custom metric groups, so the enum seems fine to me.

Presumably we could modify the metrics macros to also take a group, something like:

```
-#define POOL_COUNTER(POOL) POOL_COUNTER_PREFIX(POOL, "")
+#define POOL_COUNTER(POOL) POOL_COUNTER_PREFIX(POOL, "", DEFAULT_METRICS_GROUP)
```

btw, what kind of problems do you envision with multiple store roots? Currently, all but two tests are failing (and I think one of them is related to increasing memory limits), so the patch in its current state shouldn't be completely broken. The only variables shared between both roots is the allocator and the main thread dispatcher, so it _seems_ safe.